src/System.IO.Packaging/tests/Tests.cs

```diff
@@ -3639,6 +3639,109 @@ public void SetEmptyPropertyToNull()
+                        {
+                            Assert.Equal(fileName.Length, partStream.Length);
+                            Assert.Equal(fileName, reader.ReadToEnd());
+                        }
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        [OuterLoop]
+        public void VeryLargePart()
+        {
+            // FileAccess.Write is important, this tells ZipPackage to open the underlying ZipArchive in 
+            // ZipArchiveMode.Create mode as opposed to ZipArchiveMode.Update
+            // When ZipArchive is opened in Create it will write entries directly to the zip stream
+            // When ZipArchive is opened in Update it will write uncompressed data to memory until 
+            // the archive is closed.
+            using (var stream = new MemoryStream())
+            {
+                Uri partUri = PackUriHelper.CreatePartUri(new Uri("test.bin", UriKind.Relative));
+
+                // should compress *very well*
+                byte[] buffer =  new byte[1024 * 1024];
+                for(int i = 0; i < buffer.Length; i++)
+                {
+                    buffer[i] = (byte)(i % 2);
+                }
+                
+                const long sizeInMb = 6 * 1024; // 6GB
+                long totalLength = sizeInMb * buffer.Length;
+
+                using (Package package = Package.Open(stream, FileMode.Create, FileAccess.Write))  
+                {
+                    PackagePart part = package.CreatePart(partUri,
+                                                          System.Net.Mime.MediaTypeNames.Application.Octet,
+                                                          CompressionOption.Fast);
+
+
+                    using (Stream partStream = part.GetStream())
+                    {
+                        for (long i = 0; i < sizeInMb; i++)
+                        {
+                            partStream.Write(buffer, 0, buffer.Length);
+                        }
+                    }
+                }
+
+                // reopen for read and make sure we can get the part length & data matches
+                stream.Seek(0, SeekOrigin.Begin);
+                using (Package readPackage = Package.Open(stream))
+                {
+                    var part = readPackage.GetPart(partUri);
+
+                    using (Stream partStream = part.GetStream())
+                    {
+                        Assert.Equal(totalLength, partStream.Length);
+
+                        byte[] readBuffer = new byte[buffer.Length];
+                        for(long i = 0; i < sizeInMb; i++)
+                        {
+                            int actualRead = partStream.Read(readBuffer, 0, readBuffer.Length);
+
+                            Assert.Equal(actualRead, readBuffer.Length);
+                            Assert.True(((ReadOnlySpan<byte>)buffer).SequenceEqual(readBuffer));
```

Comment:
If you instead did:
```C#
Assert.Equal<byte>(buffer, readBuffer);
```
the failure message would provide more details about where they differed.

Reply:
When I did this before the perf was terrible.  It took over 4 minutes to compare.  I can try again.

Reply:
Ugh, if that's actually the case, stick with what you've got.

Reply:
Yeah, I retested with your suggested change.  It took over 10 minutes before I killed it.   Span extension takes about 10s on my machine.

Interestingly ReadOnlySpan.SequenceEqual was about 3x slower on desktop ~35s.  Also the entire test case is about 10x slower on desktop, so that shows the improvements in System.IO.Compression.

Reply:
Sounds like xunit's assert library could use a little perf love :)

Reply:
https://github.com/xunit/xunit/issues/1907